backend/mora/service/addresses/base.py

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (c) 2017-2018, Magenta ApS
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+import abc
+import inspect
+
+from ... import exceptions
+from ... import mapping
+from ... import util
+
+ADDRESS_HANDLERS = {}
+
+
+class _AddressHandlerMeta(abc.ABCMeta):
+    """Metaclass for automatically registering handlers."""
+
+    @staticmethod
+    def __new__(mcls, name, bases, namespace):
+        cls = super().__new__(mcls, name, bases, namespace)
+
+        if not inspect.isabstract(cls):
+            cls._register()
+
+        return cls
+
+
+class AddressHandler(metaclass=_AddressHandlerMeta):
+    __slots__ = 'scope', 'prefix', 'value'
+
+    @classmethod
+    def _register(cls):
+        ADDRESS_HANDLERS[cls.scope] = cls
+
+    def __init__(self, value):
+        self.value = value
+
+    @classmethod
+    def from_effect(cls, effect):
+        """Initialize handler from LoRa object"""
+        # Cut off the prefix
+        urn = mapping.SINGLE_ADDRESS_FIELD(effect)[0].get('urn')
+        value = urn[len(cls.prefix):]
+        return cls(value)
+
+    @classmethod
+    def from_request(cls, request):
+        """Initialize handler from MO object"""
+        value = util.checked_get(request, mapping.VALUE, "", required=True)
+        return cls(value)
+
+    def get_urn(self):
+        return self.prefix + self.value
+
+    def get_value(self):
+        return self.value
+
+    def get_pretty_value(self):
+        return self.value
+
+    def get_href(self):
+        return None
+
+    def get_lora_properties(self):
+        return []
+
+    def get_lora_address(self):
+        return {
+            'objekttype': self.scope,
+            'urn': self.get_urn(),
+        }
+
+    def get_mo_properties(self):
```

Comment:
Add a docstring; what is properties; why is it different from the address below; why are they not just one?

Also, the `_mo_` chunk is a new naming convention that does not occur anywhere else in the codebase. I think the closest is `_obj` to signify that something is, in fact, an object. However, since this is a conversion method rather than a simple accessor, an `as_` prefix might be clearer. But that's also new. If you convert the accessors to properties (or remove them), the `get_` would suffice.

Reply:
The naming was attempt to be more clear in the naming of the objects we're working with, which we haven't really made a clear distinction between so far, IMO.

We basically operate with two different groups of models. The ones we use towards LoRa, and the ones we use towards the UI.  Calling it `get_properties` wouldn't make it super obvious, that this is actually meant for the specific model used towards the UI.

Reply:
> The naming was attempt to be more clear in the naming of the objects we're working with, which we haven't really made a clear distinction between so far, IMO.

That makes sense — I'd be inclined towards something like `_api_` but `_mo_` is fine too 🙂

> We basically operate with two different groups of models. The ones we use towards LoRa, and the ones we use towards the UI. Calling it `get_properties` wouldn't make it super obvious, that this is actually meant for the specific model used towards the UI.

AFAICT you can quite safely fold `get_mo_properties()` and `get_mo_address()` into one; the only caller expands _both_ into the same dictionary, right?